andela/travel_tool_front
# #161699088 Fix reloading issues on the trip planner page.

#### What does this PR do?
Implements a fix on the trip planner page to avoid reloading when a checklist is added, edited, disabled or restored.

#### Description of Task to be completed?
* Update the action creators, reducers and sagas to get an updated record from the state without re-fetching checklists.
* Write tests for added methods.


#### How should this be manually tested?
* Clone the repo with ``` git clone https://github.com/andela/travel_tool_back.git ```
* cd to travel_tool_front
* Checkout to the branch ``` bg-fix-refresh-of-trip-planner-page-161699088 ```
* Run `` yarn ``` to install dependencies
* Start up the server with ``` yarn start ```
* Visit https://travela-local.andela.com
* Login to the application
* Navigate to the ``` Trip Planner ``` page
* Perform the following actions on the page:
    * ADD CHECKLIST
    * UPDATE CHECKLIST
    * DISABLE CHECKLIST
    * RESTORE CHECKLIST
* For any of these actions, the page should be updated without a reload.

#### Any background context you want to provide?
* You should be logged in as a ``` Travel Administrator ``` to see the Trip Planner option.

#### What are the relevant pivotal tracker stories?
[161699088](https://www.pivotaltracker.com/story/show/161699088)

#### Screenshots (if appropriate)
![add](https://user-images.githubusercontent.com/29521319/49018282-4da1d600-f18b-11e8-8f36-b84d7f037b09.gif)
![update](https://user-images.githubusercontent.com/29521319/49018298-5692a780-f18b-11e8-9e8a-358904d625e0.gif)
![disable](https://user-images.githubusercontent.com/29521319/49018324-68744a80-f18b-11e8-9d28-30af2f90c1f6.gif)
![restore](https://user-images.githubusercontent.com/29521319/49018329-6d38fe80-f18b-11e8-9482-082925d2ed5e.gif)



#### Questions:
